base_layer/core/src/transactions/key_manager/inner.rs

```diff
@@ -363,11 +365,21 @@ where TBackend: KeyManagerBackend<PublicKey> + 'static
 
         let current_index = km.key_index();
 
-        for i in 0u64..current_index + KEY_MANAGER_MAX_SEARCH_DEPTH {
-            let private_key = &km.derive_key(i)?.key;
+        // its most likely that the key is close to the current index, so we start searching from the current index
+        for i in 0u64..KEY_MANAGER_MAX_SEARCH_DEPTH {
```

Comment:
Why is `KEY_MANAGER_MAX_SEARCH_DEPTH` defined twice in the code? Can it not be defined once and included in the other location?

Reply:
Different classes. They happen to be the same number. I renamed the one to remove this confusion.